src/snowflake/snowpark/modin/plugin/_internal/resample_utils.py

```diff
@@ -715,18 +728,28 @@ def perform_asof_join_on_frame(
 
     # 4. Join left_snowpark_df and right_snowpark_df using the following logic:
     # For each element left_frame's __resample_index__, join it with a single row
-    # in right_frame whose __index__ value is less than or equal to it and is closest in time.
+    # in right_frame whose __index__ value is less/greater than or equal to it and is closest in time.
     # If a row cannot be found, pad the joined columns from right_frame with null.
-    joined_snowpark_df = left_snowpark_df.join(
-        right=right_snowpark_df,
-        on=(
+    if fill_method == "bfill":
+        on_expr = (
             left_snowpark_df[left_timecol_snowflake_quoted_identifier]
-            >= right_snowpark_df[interval_start_snowflake_quoted_identifier]
+            <= right_snowpark_df[interval_start_snowflake_quoted_identifier]
```

Comment:
Is it possible to have null values in those cases?

Reply:
We test these cases quite exhaustively in `test_resample_ffill_ser_missing_in_middle` and `test_resample_ffill_ffilled_with_none`